spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/actions/AnalyzeTableSparkAction.java

```diff
@@ -0,0 +1,184 @@
+import org.apache.iceberg.actions.ImmutableAnalyzeTable;
+import org.apache.iceberg.exceptions.ValidationException;
+import org.apache.iceberg.io.FileIO;
+import org.apache.iceberg.io.OutputFile;
+import org.apache.iceberg.puffin.Blob;
+import org.apache.iceberg.puffin.Puffin;
+import org.apache.iceberg.puffin.PuffinWriter;
+import org.apache.iceberg.puffin.StandardBlobTypes;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableSet;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.apache.iceberg.spark.JobGroupInfo;
+import org.apache.iceberg.types.Types;
+import org.apache.spark.sql.SparkSession;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Computes the statistic of the given columns and stores it as Puffin files. */
+public class AnalyzeTableSparkAction extends BaseSparkAction<AnalyzeTableSparkAction>
+    implements AnalyzeTable {
+
+  private static final Logger LOG = LoggerFactory.getLogger(AnalyzeTableSparkAction.class);
+  private static final Set<String> supportedBlobTypes =
+      ImmutableSet.of(StandardBlobTypes.APACHE_DATASKETCHES_THETA_V1);
+
+  private final Table table;
+  private Set<String> columns;
+  private Set<String> blobTypesToAnalyze = supportedBlobTypes;
+  private Long snapshotToAnalyze;
+
+  AnalyzeTableSparkAction(SparkSession spark, Table table) {
+    super(spark);
+    this.table = table;
+    Snapshot snapshot = table.currentSnapshot();
+    if (snapshot != null) {
+      snapshotToAnalyze = snapshot.snapshotId();
+    }
+    columns =
+        table.schema().columns().stream().map(Types.NestedField::name).collect(Collectors.toSet());
+  }
+
+  @Override
+  protected AnalyzeTableSparkAction self() {
+    return this;
+  }
+
+  @Override
+  public Result execute() {
+    String desc =
+        String.format("Analyzing table %s for snapshot id %s", table.name(), snapshotToAnalyze);
+    JobGroupInfo info = newJobGroupInfo("ANALYZE-TABLE", desc);
+    return withJobGroupInfo(info, this::doExecute);
+  }
+
+  private Result doExecute() {
+    List<AnalysisResult> results = Lists.newArrayList();
+    LOG.info("Starting analysis of {} for snapshot {}", table.name(), snapshotToAnalyze);
+    if (snapshotToAnalyze == null) {
+      LOG.info("Unable to analyze the table since the table has no snapshots");
+      return ImmutableAnalyzeTable.Result.builder().analyzed(false).build();
+    }
+
+    List<Blob> blobs =
+        blobTypesToAnalyze.stream()
+            .flatMap(
+                type -> {
+                  switch (type) {
+                    case StandardBlobTypes.APACHE_DATASKETCHES_THETA_V1:
+                      try {
+                        return generateNDVBlobs().stream();
+                      } catch (Exception e) {
+                        LOG.error(
```

Comment:
Since there are can be more than one type of statistic to be collected, I was think about sending the errors with respect to each stat type in the results.
But looks like this is not something very common atleast in RDBMS land.
I changed the Result to not send error message and instead throwing an Exception now